webroot/src/components/Lists/Pagination/Pagination.spec.ts

```diff
@@ -42,315 +46,115 @@ describe('Pagination component', async () => {
-    //     expect(spy.calledOnce).to.be.true;
-    // });
-    // it('should set the page to the last page when the page size change creates a larger page than exists', async () => {
-    //     const spy = sinon.spy();
-    //     const wrapper = await mountShallow(Pagination, {
-    //         props: {
-    //             paginationId: 'test3',
-    //             pageChange: spy,
-    //             listSize: 38,
-    //         }
-    //     });
-    //     const instance = wrapper.vm;
-    //
-    //     instance.setPage(4);
-    //     instance.setSize({ value: 50 });
-    //
-    //     const thirdSpy = spy.thirdCall;
-    //
-    //     expect(thirdSpy.calledWith(0, 50)).to.be.true;
-    //     expect(instance.currentPage).to.equal(1);
-    // });
+    it('should load with previous next paging', async () => {
+        const spy = sinon.spy();
+        const wrapper = await mountFull(Pagination, {
+            props: {
+                paginationId: 'test',
+                pageChange: spy,
+                pagingPrevKey: 'test-prev',
+                pagingNextKey: 'test-next',
+            },
+        });
+        const instance = wrapper.vm;
+        const pages = instance.pages.map((page) => page.displayValue);
+
+        expect(pages).to.be.an('array');
+        expect(pages).to.have.length(1);
+        expect(pages[0]).to.equal(1);
+        expect(wrapper.findComponent(LeftCaretIcon).exists(), 'previous arrow').to.equal(false);
+        expect(wrapper.findComponent(RightCaretIcon).exists(), 'next arrow').to.equal(true);
+    });
+    it('should advance with next page (2)', async () => {
+        const spy = sinon.spy();
+        const wrapper = await mountFull(Pagination, {
+            props: {
+                paginationId: 'test',
+                pageChange: spy,
+                pagingPrevKey: 'test-prev',
+                pagingNextKey: 'test-next',
+            },
+        });
+        const instance = wrapper.vm;
+
+        await wrapper.get('.pagination-item.next').trigger('click');
+
+        const pages = instance.pages.map((page) => page.displayValue);
+
+        expect(pages).to.be.an('array');
+        expect(pages).to.have.length(2);
+        expect(pages[0]).to.equal(1);
+        expect(pages[1]).to.equal(2);
+        expect(wrapper.findComponent(LeftCaretIcon).exists(), 'previous arrow').to.equal(true);
+        expect(wrapper.findComponent(RightCaretIcon).exists(), 'next arrow').to.equal(true);
+    });
+    it('should advance with next page (3)', async () => {
+        const spy = sinon.spy();
+        const wrapper = await mountFull(Pagination, {
+            props: {
+                paginationId: 'test',
+                pageChange: spy,
+                pagingPrevKey: 'test-prev',
+                pagingNextKey: 'test-next',
+            },
+        });
+        const instance = wrapper.vm;
+
+        await wrapper.get('.pagination-item.next').trigger('click');
```

Comment:
Do these tests compound? Otherwise, why would there be 3 pages from a single click?

Reply:
Yep, the mock store is shared in these cases.